realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -360,6 +360,18 @@ public DynamicRealmObject getObject(String fieldName) {
     }
 
     /**
+     * Returns the {@link RealmList} of values being linked from the given field.
+     *
+     * @param fieldName the name of the field.
+     * @return the {@link RealmList} data for this field.
+     * @throws IllegalArgumentException if field name doesn't exist or it doesn't contain a list of values.
+     */
+    public <T> RealmList<T> getValueList(String fieldName, Class<T> valueClass) {
```

Comment:
We need to know which class is actually used as type parameter of `RealmList`.
When the list field is INTEGER_LIST in schema, `T` can be `Long`, `Integer`, `Short` and `Byte`.


```
DynamicRealmObject obj = ...;
RealmList<Short> list = obj.getList("shortList");

Short value = list.get(0); // the implementation of 'get(int)' must return `Short` object.
```